services/openshiftmisc/src/handlers/resticRestore.js

```diff
@@ -0,0 +1,97 @@
+// @flow
+
+const promisify = require('util').promisify;
+const OpenShiftClient = require('openshift-client');
+const R = require('ramda');
+const { logger } = require('@lagoon/commons/src/local-logging');
+const { sendToLagoonLogs } = require('@lagoon/commons/src/logs');
+const {
+  getOpenShiftInfoForProject,
+} = require('@lagoon/commons/src/api');
+
+
+async function resticRestore (data: Object) {
+  const { backup, restore, project, environment } = data;
+
+  const result = await getOpenShiftInfoForProject(project.name);
+  const projectOpenShift = result.project;
+
+  const ocsafety = string =>
+    string.toLocaleLowerCase().replace(/[^0-9a-z-]/g, '-');
+
+  try {
+    var safeBranchName = ocsafety(environment.name);
+    var safeProjectName = ocsafety(project.name);
+    var openshiftConsole = projectOpenShift.openshift.consoleUrl.replace(
+      /\/$/,
+      ''
+    );
+    var openshiftToken = projectOpenShift.openshift.token || '';
+    var openshiftProject = projectOpenShift.openshiftProjectPattern
+      ? projectOpenShift.openshiftProjectPattern
+          .replace('${branch}', safeBranchName)
+          .replace('${project}', safeProjectName)
+      : `${safeProjectName}-${safeBranchName}`;
+    var restoreName = `restore-${R.slice(0, 7, backup.backupId)}`;
+  } catch (error) {
+    logger.error(`Error while loading information for project ${project.name}`);
+    logger.error(error);
+    throw error;
+  }
+
+  const restoreConfig = (name, backupId) => {
+    let config = {
+      apiVersion: 'appuio.ch/v1alpha1',
+      kind: 'Restore',
+      metadata: {
+        name
+      },
+      spec: {
+        snapshot: backupId,
+        restoreMethod: {
+          s3: {},
+        },
+        backend: {
+          s3: {},
+        },
+      },
+    };
+
+    return config;
+  };
+
+  // Kubernetes API Object - needed as some API calls are done to the Kubernetes API part of OpenShift and
+  // the OpenShift API does not support them.
+  const kubernetes = new OpenShiftClient.Core({
+    url: openshiftConsole,
+    insecureSkipTlsVerify: true,
+    auth: {
+      bearer: openshiftToken
+    }
+  });
+
+  try {
+    const restoreConfigPost = promisify(
+      kubernetes.ns(openshiftProject).restore.post
```

Comment:
I'm highly confident that this doesn't work as is. Minishift doesn't have this api, and the [appuio docs](http://docs.appuio.ch/en/latest/baas.html#) don't mention api support either.

Need to find out what the api path is, if it exists, or do something altogether different.

Reply:
Fixed and tested locally!